controller/eks-cluster-config-handler.go

```diff
@@ -819,6 +827,7 @@ func BuildUpstreamClusterState(name, managedTemplateID string, clusterState *eks
 			Tags:                 ng.Nodegroup.Tags,
 			Version:              ng.Nodegroup.Version,
 			RequestSpotInstances: aws.Bool(aws.StringValue(ng.Nodegroup.CapacityType) == eks.CapacityTypesSpot),
+			NodeRole:             ng.Nodegroup.NodeRole,
```

Comment:
A nodegroup will always be created with a role: either a custom one or the one created by Rancher.

With this in mind, we need to detect that here and not set the `NodeRole` field if the `NodeRole` on the nodegroup is the one created by Rancher. There is a specific reason for this related to syncing that is done on the Rancher side.